nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -318,6 +319,8 @@ class Machine:
     shell: Optional[socket.socket]
     serial_thread: Optional[threading.Thread]
 
+    tpm: Tpm | None
```

Comment:
```suggestion
    tpm: Optional[Tpm]
```
Not a python typing expert here, but that's what `Optional` is about right?

Reply:
The Union `|` Operator is newer than `Optional`. It was introduced in Python 3.10 ([PEP 604](https://peps.python.org/pep-0604/)) and I'd wager, nowadays, `Tpm | None` is even more idiomatic than the older `Optional[Tpm]`.